drv/cosmo-hf/src/hf.rs

```diff
@@ -50,6 +68,134 @@ impl ServerImpl {
     fn flash_addr(&self, offset: u32) -> u32 {
         offset + self.flash_base()
     }
+
+    /// Scans the currently active (virtual) device for persistent data
+    ///
+    /// Returns a tuple containing
+    /// - newest persistent data record
+    /// - address of first empty slot
+    fn persistent_data_scan(
+        &mut self,
+    ) -> (Option<HfRawPersistentData>, Option<u32>) {
+        let mut best: Option<HfRawPersistentData> = None;
+        let mut empty_slot: Option<u32> = None;
+        for i in 0..SECTOR_SIZE_BYTES / HF_PERSISTENT_DATA_STRIDE as u32 {
+            let addr = i * HF_PERSISTENT_DATA_STRIDE as u32;
+            let mut data = HfRawPersistentData::new_zeroed();
+            self.drv
+                .flash_read(self.flash_addr(addr), &mut data.as_bytes_mut())
+                .unwrap_lite(); // flash_read is infallible when using a slice
+            if data.is_valid() && best.map(|b| data > b).unwrap_or(true) {
```

Comment:
**thought:** I spent so long pondering: "Why on earth is bigger better?" :D Took me way too long to scroll down and find that there's a custom Ord impl.